internal/httputil/server.go

```diff
@@ -98,15 +90,35 @@ func newDefaultTLSConfig(cert *tls.Certificate) *tls.Config {
 	return tlsConfig
 }
 
-// grpcHandlerFunc splits request serving between gRPC and HTTPS depending on
-// the request type. Requires HTTP/2 to be enabled.
-func grpcHandlerFunc(rpcServer http.Handler, other http.Handler) http.Handler {
+// RedirectHandler takes an incoming request and redirects to its HTTPS counterpart
+func RedirectHandler() http.Handler {
 	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
-		ct := r.Header.Get("Content-Type")
-		if r.ProtoMajor == 2 && strings.Contains(ct, "application/grpc") {
-			rpcServer.ServeHTTP(w, r)
-		} else {
-			other.ServeHTTP(w, r)
-		}
+		w.Header().Set("Connection", "close")
+		url := fmt.Sprintf("https://%s%s", urlutil.StripPort(r.Host), r.URL.String())
```

Comment:
Will this handle alternate ports correctly?

Reply:
No, it will only redirect to the default https.

More concerning, in adding some unit tests to this function... I'm not sure this has ever worked as intended.

Reply:
:sadtrombone:

Reply:
Cool if we punt on this for now? It works as (originally) intended but not for non `:443` redirects.

Reply:
Yeah no sense adding to the scope here.